fizz-buzz/Readme.md

```diff
@@ -0,0 +1,115 @@
+# CSC207 Course Notes - Fizz Buzz
+
+Let's start off our exploration of Java with a classic programming challenge, Fizz Buzz!
+
+## Fizz Buzz
+
+If you have not heard of this, Fizz Buzz is a game where we must create different outputs based on a number we continually increment.
+Considering the current number is `i`, we output the following:
+
+* If `i` is divisible by 3, print 'Fizz'
+* If `i` is divisible by 5, print 'Buzz'
+* If `i` is divisibly by 3 and 5, print 'Fizz Buzz'
+* Otherwise, print the value of `i`
+
+So, starting with `i = 1` and counting until `i = 6` we would get:
+
+```
+1
+2
+Fizz
+4
+Buzz
+Fizz
+```
+
+Now, we want to create a program to do this for us.
+We could write it in Python (and encourage that you quickly try this if you have never before) but, we will present the program written in Java and dissect it.
+Fizz Buzz, implemented in Java may look like:
+
+```java
+class FizzBuzz {
+
+    public static void main(String[] args) {
+
+        for (int i = 1; i < 100; i++) {
+
+            // Find out which numbers divide i.
+            boolean divisible_by_3 = i % 3 == 0;
+            boolean divisible_by_5 = i % 5 == 0;
+
+            // Print our appropriate result.
+            if (divisible_by_3 && divisible_by_5) {
+
+                System.out.println("Fizz Buzz");
+
+            } else if (divisible_by_3) {
+
+                System.out.println("Fizz");
+
+            } else if (divisible_by_5) {
+
+                System.out.println("Buzz");
+
+            } else {
+
+                System.out.println(i);
+
+            }
+
+        }
+
+    }
+
+}
+```
+
+This program (which you should run for yourself) prints out the first 100 numbers of Fizz Buzz!
+Now, you may never have seen Java code before but still, just by reading over what is written here, you may be able to get a feel for how Java code works.
+Take a minute and read this through and take guesses at what different pieces of the code do.
```

Comment:
What's going with Java types? How does that weird for loop work? When do you use semicolons and curly braces? Is that really how you print in Java -- what the heck were the language designers thinking?!?